### ISSUE TITLE
"Cannot convert value to type Nullable`1.

### ISSUE THREAD

**Issue by captainDeezN** [mrxten/db-request-filters#4]
Getting the above error when  trying to filter using a property in my model class that is of type int.

Below is my filter declaration:

```
var filter = new FilterContainer
      {
        Where = new TreeFilter
        {
          OperatorType = TreeFilterType.Or,
          Operands = new List<TreeFilter>
                    {
                        new TreeFilter
                        {
                            Field = "Description",
                            FilterType = WhereFilterType.Contains,
                            Value = "test"
                        },
                          new TreeFilter
                        {
                            Field = "QAApproverId",
                            FilterType = WhereFilterType.Equal,
                            Value = 2
                            
                        },


                    }
        },

        OrderBy = new List<OrderFilter>
                {
                    new OrderFilter
                    {
                        Field = "Id",
                        Order = OrderFilterType.Desc
                    },
                }
      };

```
And this is a piece of my Model Class:

 public int? QAApproverId { get; set; }
 [ForeignKey("QAApproverId")]
 public User QAApprover { get; set; }

**Comment by mrxten** [comment]
Hi, will fix it.

**Comment by BramscoChill** [comment]
was having this same issue, I am using the entity framework in combination with QueryDesigner. When there are properties of the nullable type in the model, its crashing when filtering with:  "System.InvalidCastException: 'Cannot convert value to type Nullable`1.'" inside: WhereExpression.cs --> GenerateExpressionOneField() on: Expression.Equal(). Inside Expression.Equal(), both expressions need to be of the same type. I managed to fix this with the following (in: WhereExpression.cs):

[WhereExpression_FIX.txt](https://github.com/mrxten/QueryDesigner/files/1851391/WhereExpression_FIX.txt)

**Comment by mrxten** [comment]
@BramscoChill wow, sorry for late answers, too busy :(
Thanks for contribution, am will check this and will add to library :)

**Comment by mrxten** [comment]
Fixed in 2.1.0-pre
